client/src/pages/contentPages/ForumSidePanel.js

```diff
@@ -0,0 +1,120 @@
+import React, { useState } from 'react';
+import { Grid, Typography, Badge, Button } from '@material-ui/core';
+import { makeStyles } from '@material-ui/core/styles';
+import AddIcon from '@material-ui/icons/Add';
+import RemoveIcon from '@material-ui/icons/Remove';
+import { courseGroupList } from '../../data/mockData';
+
+const useStyles = makeStyles(theme => ({
+    chat_head: {
+        height: 120,
+        display: 'flex',
+        justifyContent: 'flex-start',
+        alignItems: 'center',
+        paddingLeft: theme.spacing(4),
+        '& p': {
+            fontSize: '1.375rem',
+        },
+    },
+    badge: {
+        color: theme.palette.common.white,
+        paddingLeft: theme.spacing(4),
+        '& span': {
+            width: 40,
+            height: 24,
+            fontSize: '0.875rem',
+            background: theme.palette.primary.gradient,
+        },
+    },
+    accordion: {
+        display: 'flex',
+        flexDirection: 'column',
+    },
+    accordion_container: {
+        display: 'block',
+        width: '85%',
+        backgroundColor: '#e6edff',
+        margin: theme.spacing(0, 5, 2, 4),
+        padding: theme.spacing(2.5),
+        borderRadius: 8,
+    },
+    course_name: {
+        display: 'flex',
+        justifyContent: 'space-between',
+        alignItems: 'center',
+        '& p': {
+            fontSize: '1.125rem',
+        },
+    },
+    icons: {
+        color: '#2968ff',
+    },
+    group_list: {
+        color: theme.palette.primary.main,
+        cursor: 'pointer',
+        marginTop: 16,
+    },
+}));
+
+const ForumSidePanel = () => {
+    const classes = useStyles();
+
+    const [courseList, setCourseList] = useState(
+        courseGroupList.map(cgl => ({ ...cgl, expand: false }))
+    );
+    const [courseId, setCourseId] = useState([]);
+
+    const showGroup = id => {
+        setCourseId([...courseId, id]);
+        const editedCourseList = courseList.map(cl => {
+            if (cl.id === id && !cl.expand) return { ...cl, expand: true };
+            if (cl.id === id && cl.expand) return { ...cl, expand: false };
+            else return { ...cl };
+        });
```

Comment:
I also think the logic here works best inside the component so they're all controlling their own toggle boolean. If it's outside you have to pass it the id. Whereas if the logic and the variable are scoped inside the component. In theory they should only be able to control their own boolean to show or hide because the variable will only be accessed inside the component level.

Reply:
Good point, I will try to use descriptive variable names. I'll always try to do that but sometimes unique descriptive names become long.